atef/widgets/config.py

```diff
@@ -1865,16 +1865,30 @@ def eventFilter(self, object: QLineEdit, event: QEvent) -> bool:
         if not isinstance(object, QLineEdit):
             return False
         if event.type() == QEvent.FocusIn:
-            object.setFrame(True)
-            object.setReadOnly(False)
+            self.set_edit_style(object)
             return True
         if event.type() == QEvent.FocusOut:
-            if object.text():
-                object.setFrame(False)
-            object.setReadOnly(True)
+            self.set_no_edit_style(object)
             return True
         return False
 
+    @staticmethod
+    def set_edit_style(object: QLineEdit):
+        object.setFrame(True)
+        object.setStyleSheet(
+            "QLineEdit { background: white }"
```

Comment:
(I think this is good as-is, just wondering aloud...)

I wonder if this would be possible to do in a single QSS block with 2 state-based styles (https://doc.qt.io/qt-5/stylesheet-reference.html; active vs non-active or something). 

The read-only part definitely wouldn't be, but the frame/background may be.

Reply:
So, instead of using a custom filter on focus events for the background, the alternative here is to use a pseudo-state selector to set the background style? e.g., "in this state, background transparent" (in other states, no change).
For example, perhaps `QLineEdit[:no_frame] { background: transparent }`

Reply:
I tried for a bit to get this to work but I can't figure out the precise invocation and stylesheet debug is pain. Skipping (but still editing to use the proper palette color)